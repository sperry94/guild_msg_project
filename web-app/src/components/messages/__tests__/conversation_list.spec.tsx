import React from "react"
import {
  useConversationList,
  Conversation,
} from "../../../services/conversations"
import renderer from "react-test-renderer"
import ConversationList, { ConversationListProps } from "../conversation_list"

jest.mock("../../../services/auth")
jest.mock("../../../services/firebase")
jest.mock("../../../services/conversations")
jest.mock("../../../services/messages")

const mocked_useConversationList = useConversationList as jest.Mock<
  Conversation[] | null
>

describe("ConversationList", () => {
  const cases: ConversationListProps[] = [
    {
      conversationId: undefined,
    },
    {
      conversationId: "convo-id",
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1} with conversations`, () => {
      mocked_useConversationList.mockReturnValueOnce([
        {
          id: "convo-id",
          users: [
            {
              displayName: "Bob",
              email: "bob@bob.com",
              uid: "uid1",
              photoUrl: "",
            },
          ],
        },
        {
          id: "convo-id2",
          users: [
            {
              displayName: "Bill",
              email: "bill@bob.com",
              uid: "uid2",
              photoUrl: "",
            },
          ],
        },
      ])
      const tree = renderer.create(<ConversationList {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`renders for case #${index + 1} without conversations`, () => {
      mocked_useConversationList.mockReturnValueOnce([])
      const tree = renderer.create(<ConversationList {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`renders for case #${index + 1} while loading`, () => {
      mocked_useConversationList.mockReturnValueOnce(null)
      const tree = renderer.create(<ConversationList {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
