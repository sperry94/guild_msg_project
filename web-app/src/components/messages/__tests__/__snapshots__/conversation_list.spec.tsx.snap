// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationList renders for case #1 while loading 1`] = `
<div
  className="MuiCircularProgress-root makeStyles-spinner-3 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ConversationList renders for case #1 with conversations 1`] = `
<nav
  className="makeStyles-nav-1"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-link-8 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        href="/messages/convo-id"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
        >
          B
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          Bob
        </p>
      </a>
    </li>
    <li>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-link-8 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        href="/messages/convo-id2"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
        >
          B
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          Bill
        </p>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`ConversationList renders for case #1 without conversations 1`] = `
<p
  className="MuiTypography-root makeStyles-noConversationText-2 MuiTypography-h6"
>
  Add a conversation below
</p>
`;

exports[`ConversationList renders for case #2 while loading 1`] = `
<div
  className="MuiCircularProgress-root makeStyles-spinner-3 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ConversationList renders for case #2 with conversations 1`] = `
<nav
  className="makeStyles-nav-1"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-link-8 MuiListItem-gutters MuiListItem-divider MuiListItem-button Mui-selected"
        href="/messages/convo-id"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
        >
          B
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          Bob
        </p>
      </a>
    </li>
    <li>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-link-8 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        href="/messages/convo-id2"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
        >
          B
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          Bill
        </p>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`ConversationList renders for case #2 without conversations 1`] = `
<p
  className="MuiTypography-root makeStyles-noConversationText-2 MuiTypography-h6"
>
  Add a conversation below
</p>
`;
